modules/nlu/src/backend/pipelines/intents/svm_classifier.ts

```diff
@@ -127,18 +127,26 @@ export default class SVMClassifier {
     for (const [index, context] of Object.entries(allContexts)) {
       const intents = intentsWTokens.filter(x => x.contexts.includes(context))
       const utterances = _.flatten(intents.map(x => x.tokens))
+      const vocabulary = _.flatten(utterances)
+
+      // Vectorize all the vocabulary upfront to batch and cache it
+      await this.languageProvider.vectorize(vocabulary, this.language)
 
       // Generate 'none' utterances for this context
-      const junkWords = await this.languageProvider.generateSimilarJunkWords(_.flatten(utterances))
-      const nbOfNoneUtterances = Math.max(5, utterances.length / 2) // minimum 5 none utterances per context
-      const noneUtterances = _.range(0, nbOfNoneUtterances).map(() => _.sampleSize(junkWords))
+      const junkWords = await this.languageProvider.generateSimilarJunkWords(vocabulary, this.language)
+      const nbOfNoneUtterances = Math.max(5, utterances.length / 2) // minimum 5 none utterances per context, max 50% of the utterances set
```

Comment:
don't know if this is on purpose but you wrote minimum of 5 in the comment and yet you're using Mat.max